code/modules/client/preferences/preferences_savefile.dm

```diff
@@ -0,0 +1,609 @@
+/// IF YOU NEED A FIELD ADDED TO THE DATABASE, CREATE A MIGRATION SO SHIT GETS UPDATED.
+/// Also update SQL/players2.sql.
+/// SEE code/modules/migrations/SS13_Prefs/
+
+/datum/preferences/proc/SetChangelog(ckey,hash)
+	var/datum/preference_setting/lastchangelog = get_pref_datum(/datum/preference_setting/string/changelog)
+	lastchangelog.setting=hash
+	var/database/query/q = new
+	q.Add("UPDATE client SET lastchangelog=? WHERE ckey=?",lastchangelog.setting,ckey)
+	if(!q.Execute(db))
+		message_admins("Error in SetChangelog [__FILE__] ln:[__LINE__] #: [q.Error()] - [q.ErrorMsg()]")
+		WARNING("Error in Setchangelog [__FILE__] ln:[__LINE__] #:[q.Error()] - [q.ErrorMsg()]")
+		return 0
+
+
+/datum/preferences/proc/try_load_preferences(var/ckey, var/mob/user)
+	// Check client:
+	var/database/query/client_check = new
+	client_check.Add("SELECT ckey FROM client WHERE ckey = ?", ckey)
+	if(client_check.Execute(db))
+		if(!client_check.NextRow())
+			return save_preferences_sqlite(ckey, user)
+		else
+			return load_preferences_sqlite(ckey)
+	else
+		WARNING("Error in try_load_preferences [__FILE__] ln:[__LINE__] #:[client_check.Error()] - [client_check.ErrorMsg()]")
+
+/datum/preferences/proc/load_preferences_sqlite(var/ckey)
+	var/list/database_data = execute_load_pref_query(ckey)
+	read_database_data_client(database_data)
+	initialize_preferences()
+	if (islist(database_data))
+		return 1
+	else
+		return 0
+
+/datum/preferences/proc/execute_load_pref_query(var/ckey)
+	var/list/preference_list_client = new
+	var/database/query/check = new
+	var/database/query/q = new
+	check.Add("SELECT ckey FROM client WHERE ckey = ?", ckey)
+	if(check.Execute(db))
+		if(!check.NextRow())
+			message_admins("Empty client setting for [ckey]!")
+			WARNING("Empty client setting for [ckey]")
+			return 0
+	else
+		message_admins("Error in load_preferences_sqlite [__FILE__] ln:[__LINE__] #: [check.Error()] - [check.ErrorMsg()]")
+		WARNING("Error in load_preferences_sqlite [__FILE__] ln:[__LINE__] #:[check.Error()] - [check.ErrorMsg()]")
+		return 0
+	q.Add("SELECT * FROM client WHERE ckey = ?", ckey)
+	if(q.Execute(db))
+		while(q.NextRow())
+			var/list/row = q.GetRowData()
+			for(var/a in row)
+				preference_list_client[a] = row[a]
+	else
+		message_admins("Error in load_preferences_sqlite [__FILE__] ln:[__LINE__] #: [q.Error()] - [q.ErrorMsg()]")
+		WARNING("Error in load_preferences_sqlite [__FILE__] ln:[__LINE__] #:[q.Error()] - [q.ErrorMsg()]")
+		return 0
+	return preference_list_client
+
+/datum/preferences/proc/read_database_data_client(var/list/database_data)
+	for (var/key, value in preference_settings_client)
+		var/datum/preference_setting/setting_datum = value
+		try
```

Comment:
I try/catch these to get a `CRASH()` output with more data. This was mostly for debug, but I think it's good to keep in it in case someone wants to improve on this rework.